src/librustc_mir/build/mod.rs

```diff
@@ -341,6 +341,11 @@ impl<'a,'tcx> Builder<'a,'tcx> {
             }
         }
     }
+
+    pub fn check_overflow(&self) -> bool {
```

Comment:
@eddyb well how overflow checking works with generics/inlined functions has always been unclear. This behaviour is at least more consistent with non-inlined functions.